includes/customer-functions.php

```diff
@@ -623,6 +623,13 @@ function edd_maybe_add_customer_address( $customer_id = 0, $data = array() ) {
 		return false;
 	}
 
+	// Temporarily remove the name so we can see if the address is empty.
+	$name = $data['name'];
+	unset( $data['name'] );
```

Comment:
```suggestion
	unset( $data['name'] );
	$data = array_filter( $data );
```

This will remove any empty values that have been passed through. Without this, the address would still be submitted if empty values were provided like this:

```php
$address = edd_maybe_add_customer_address( 1, array(
	'address' => '',
	'address2' => '',
	'country' => '',
	'region' => ''
) );
```
`array_filter()` removes any empty keys. So if they're all empty strings like that, we'll end up with a truly empty array and then the address won't be saved.